### PR TITLE
Added `skipOuterContainers` option

### PR DESCRIPTION
**Pull request by SilverFire**
This PR adds `skipOuter` option.

The use case:

```html
<div class="container" id="pjax-container">
    Go to <a href="/page/2">next page</a>.
    <div class="status-container" id="pjax-status-container">
        <span>Last check found no errors on page 1</span>
        <a href="/update-status/page/1">Check again</a>
    </div>
</div>
<script>
    $(document).pjax('a', '#pjax-container');
    $(document).pjax('a', '#pjax-status-container', {
        push: false,
        replace: false,
        timeout: 0
    })
</script>
```

I want to use a link inside `div#pjax-container` to navigate between pages with pushing URLs to the history. Every page displays some info and has a button `Check again` to update that info without page refresh or pushing to the browser history. As you see, I placed the button inside new container and defined a special config for it.

Without out this PR, the config of `#pjax-status-container` is useless, because the network request of `#pjax-status-container` will be immediately aborted by `#pjax-container`. Then `#pjax-container` will create his own request and process it according to it's rules.

When `skipOuter = true` only the closest container will handle the request.

The options defaults to false, so it doesn't break BC.

### CODE REVIEW

**Comment by SilverFire** [comment]
@nkovacs I know you are familiar with pjax. Could you take a look, please?

**Comment by samdark** [comment]
Name sounds a bit too vague to me. The logic of the change is OK.

**Comment by SilverFire** [comment]
There were two other names, that I tried:
 - `skipOuterPjax`: we are configuring pjax, there's no necessity in `pjax` tie
 - `skipOuterContaiter`: clear, but long

When anyone has better name - please, text it!

**Comment by samdark** [comment]
`skipOuterContaiter` sounds much better to me even if it's longer.

**Comment by SilverFire** [comment]
Okay, let it be `skipOuterContaiters`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
-2.0.1 Oct 10, 2014
-------------------
-- Bug #9: Fixed missing history option in default settings (tonydspaniard)
-- New #11: add new option "cache" (macklay)
-
-
-2.0.0 Mar 20, 2014
-------------------
-- Bug: Fixed avoid duplicates of _pjax parameter (tof06)
-- Bug: Fixed Pjax/GridView and back button (klevron, tof06, tonydspaniard)
```
Comment (samdark) [inline CHANGELOG.md]: What happened here?

Comment (SilverFire) [inline CHANGELOG.md]: `\r\n` -> `\n`

Comment (samdark) [inline CHANGELOG.md]: OK/

**Comment by nkovacs** [comment]
There are some fixes to nested containers and pjax's cache upstream. It's also possible Yii's widget is initializing the pjax code incorrectly.

**Comment by SilverFire** [comment]
> There are some fixes to nested containers

Yeah, it's time to merge with origin. @samdark  ?

> It's also possible Yii's widget is initializing the pjax code incorrectly.

Why?

**Comment by nkovacs** [comment]
There are various ways to configure pjax, e.g. you can add data-pjax="container-selector" on links to specify a container instead of doing it in the $.pjax call. Maybe some combination of these makes nested containers work, and another breaks them.
I'd try reproducing this with the upstream version, without Yii, first.

**Comment by nkovacs** [comment]
Looking at the pjax code: 
it calls preventDefault:
https://github.com/defunkt/jquery-pjax/blob/master/jquery.pjax.js#L98
and it checks for isDefaultPrevented: https://github.com/defunkt/jquery-pjax/blob/master/jquery.pjax.js#L83

So the outer container should receive the event with isDefaultPrevented, and abort.

**Comment by SilverFire** [comment]
@nkovacs You think we should call `event.preventDefault()` instead on `return` [here](https://github.com/yiisoft/jquery-pjax/pull/29/files#diff-256b2e7d3ffe31142c00fa086e15d3a6R379)?

**Comment by SilverFire** [comment]
Will merge this if nobody complains

**Comment by SilverFire** [comment]
@nkovacs I checked it. It's not the same.

**Comment by SilverFire** [comment]
https://github.com/yiisoft/jquery-pjax/commit/ef12a6bddadbb64445b3001821cfa0314a246fe7

**Pull request closed without merging**